src/main/java/io/kestra/plugin/aws/runner/AwsBatchScriptRunner.java

```diff
@@ -0,0 +1,546 @@
+            }
+        }
+
+        return new RunnerResult(0, logConsumer);
+    }
+
+    private void cleanupS3Resources(RunContext runContext, List<String> filesToUpload, List<String> filesToDownload, String workingDirKey, String outputDirKey) throws IllegalVariableEvaluationException {
+        String renderedBucket = runContext.render(s3Bucket);
+        try(S3AsyncClient s3AsyncClient = asyncClient(runContext)) {
+            List<CompletableFuture<DeleteObjectsResponse>> deletions = new ArrayList<>();
+            if (filesToUpload != null && !filesToUpload.isEmpty()) {
+                deletions.add(s3AsyncClient.deleteObjects(
+                    DeleteObjectsRequest.builder()
+                        .bucket(renderedBucket)
+                        .delete(Delete.builder()
+                            .objects(
+                                filesToUpload.stream()
+                                    .map(relativePath -> ObjectIdentifier.builder()
+                                        .key(workingDirKey + "/" + relativePath)
+                                        .build()
+                                    )
+                                    .toList()
+                            ).build()
+                        ).build()
+                ));
+            }
+
+            if (filesToDownload != null && !filesToDownload.isEmpty()) {
+                deletions.add(s3AsyncClient.deleteObjects(
+                    DeleteObjectsRequest.builder()
+                        .bucket(renderedBucket)
+                        .delete(Delete.builder()
+                            .objects(
+                                filesToDownload.stream()
+                                    .map(relativePath -> ObjectIdentifier.builder()
+                                        .key(outputDirKey + "/" + relativePath)
+                                        .build()
+                                    )
+                                    .toList()
+                            ).build()
+                        ).build()
+                ));
+            }
+
+            deletions.stream().parallel().forEach(throwConsumer(CompletableFuture::get));
+        } catch (Exception e) {
+            runContext.logger().warn("Error while cleaning up S3: {}", e.getMessage());
+        }
+    }
+
+    private S3TransferManager transferManager(RunContext runContext) throws IllegalVariableEvaluationException {
+        return S3TransferManager.builder()
+            .s3Client(this.asyncClient(runContext))
+            .build();
+    }
+
+    private static StartLiveTailResponseHandler getStartLiveTailResponseStreamHandler(Logger logger, AbstractLogConsumer logConsumer) {
+        return StartLiveTailResponseHandler.builder()
+            .onError(throwable -> {
+                CloudWatchLogsException e = (CloudWatchLogsException) throwable.getCause();
+                logger.error(e.awsErrorDetails().errorMessage());
```

Comment:
Maybe use the logConsumer instead of the runcontext logger wdyt?

Reply:
You're right, my bad :+1: